js/data/buffer/collision_box_vertex_buffer.js

```diff
@@ -1,37 +1,34 @@
 'use strict';
 
 var util = require('../../util/util');
-var Buffer = require('./buffer');
-
-module.exports = CollisionBoxVertexBuffer;
-
-function CollisionBoxVertexBuffer(buffer) {
-    Buffer.call(this, buffer);
+var Buffer2 = require('../buffer2');
+
+function CollisionBoxVertexBuffer(options) {
+    Buffer2.call(this, options || {
+        type: Buffer2.BufferType.VERTEX,
+        attributes: {
+            verticies: {
+                components: 6, // bytes per vertex (2 * short + 1 * short + 2 * byte = 8 bytes)
+                type: Buffer2.AttributeType.UNSIGNED_SHORT
+            }
+        }
+    });
 }
 
-CollisionBoxVertexBuffer.prototype = util.inherit(Buffer, {
-    itemSize: 12, // bytes per vertex (2 * short + 1 * short + 2 * byte = 8 bytes)
-    defaultLength: 32768,
-
+CollisionBoxVertexBuffer.prototype = util.inherit(Buffer2, {
     // add a vertex to this buffer;
     // x, y - vertex position
     // ex, ey - extrude normal
     add: function(point, extrude, maxZoom, placementZoom) {
-        var pos = this.pos,
-            pos2 = pos / 2,
-            index = this.index;
-
-        this.resize();
-
-        this.shorts[pos2 + 0] = point.x;
-        this.shorts[pos2 + 1] = point.y;
-
-        this.shorts[pos2 + 2] = Math.round(extrude.x);
-        this.shorts[pos2 + 3] = Math.round(extrude.y);
-        this.ubytes[pos + 8] = Math.floor(maxZoom * 10);
-        this.ubytes[pos + 9] = Math.floor(placementZoom * 10);
-
-        this.pos += this.itemSize;
-        return index;
+        this.push([
```

Comment:
... and then this would look like 

```js
this.push({
  coordinates: [ point.x, point.y ],
  extrusions: [ Math.round(extrude.x), Math.round(extrude.y) ],
  maxZoom: Math.floor(maxZoom * 10),
  placementZoom: Math.floor(placementZoom * 10)
})
```